tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -0,0 +1,117 @@
+// Author: Vincenzo Eduardo Padulano CERN/UPV 05/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RDF_RDATASETSPEC
+#define ROOT_RDF_RDATASETSPEC
+
+#include <string>
+#include <vector>
+#include <limits>
+#include <stdexcept> // std::logic_error
+#include <algorithm> // std::transform
+
+#include <RtypesCore.h>
+#include <ROOT/InternalTreeUtils.hxx>
+
+namespace ROOT {
+
+namespace RDF {
+
+struct RDatasetSpec {
+
+   struct REntryRange {
+      ULong64_t fStartEntry{0};
+      ULong64_t fEndEntry{std::numeric_limits<ULong64_t>::max()};
+      REntryRange() {}
+      REntryRange(ULong64_t endEntry) : fEndEntry(endEntry) {}
+      REntryRange(ULong64_t startEntry, ULong64_t endEntry)
+         : fStartEntry(startEntry),
+           fEndEntry(endEntry >= startEntry
+                        ? endEntry
+                        : throw std::logic_error("RDatasetSpec: fStartEntry cannot be larger than fEndEntry."))
+      {
+      }
+   };
+
+   /**
+    * A list of names of trees.
+    * This list should go in lockstep with fFileNameGlobs, only in case this dataset is a TChain where each file
+    * contains its own tree with a different name from the global name of the dataset.
+    */
+   std::vector<std::string> fTreeNames{};
+
+   /**
+    * A list of file names.
+    * They can contain the globbing characters supported by TChain. See TChain::Add for more information.
+    */
+   std::vector<std::string> fFileNameGlobs{};
+
+   ULong64_t fStartEntry{}; ///< The entry where the dataset processing should start (inclusive).
+   ULong64_t fEndEntry{};   ///< The entry where the dataset processing should end (exclusive).
+
+   ROOT::Internal::TreeUtils::RFriendInfo fFriendInfo{}; ///< List of friends
+
+   RDatasetSpec(const std::string &treeName, const std::string &fileName, REntryRange entryRange = {})
+      : fTreeNames(std::vector<std::string>{std::move(treeName)}),
+        fFileNameGlobs(std::vector<std::string>{std::move(fileName)}), fStartEntry(entryRange.fStartEntry),
+        fEndEntry(entryRange.fEndEntry)
+   {
+   }
+
+   RDatasetSpec(const std::string &treeName, const std::vector<std::string> &fileNames, REntryRange entryRange = {})
```

Comment:
See comment to first constructor, the string parameter maybe could passes as `std::string_view`
```suggestion
   RDatasetSpec(const std::string &treeName, const std::vector<std::string> &fileNameGlobs, REntryRange entryRange = {})
```

Reply:
```suggestion
   RDatasetSpec(const std::string &treeName, const std::vector<std::string> &fileNames, const REntryRange &entryRange = {})
```